docs/how-to/performing-causal-profiling.rst

```diff
@@ -0,0 +1,630 @@
+|                  |                                     |                                  | experiments                                |
++------------------+-------------------------------------+----------------------------------+--------------------------------------------+
+| Source scope     | ``OMNITRACE_CAUSAL_SOURCE_SCOPE``   | regular expression(s)            | ``<file>`` and/or ``<file>:<line>``        |
+|                  |                                     |                                  | containing code to include in experiments  |
++------------------+-------------------------------------+----------------------------------+--------------------------------------------+
+| Function scope   | ``OMNITRACE_CAUSAL_FUNCTION_SCOPE`` | regular expression(s)            | Restricts experiments to matching          |
+|                  |                                     |                                  | functions (function mode) or lines of      |
+|                  |                                     |                                  | code within matching functions (line mode) |
++------------------+-------------------------------------+----------------------------------+--------------------------------------------+
+
+.. note::
+
+   * Binary scope defaults to ``%MAIN%`` (in the executable), but the scope can be expanded to include linked libraries.
+   * ``<file>`` and ``<file>:<line>`` support requires debug info (for example, the code must be compiled with ``-g`` or, preferably, with ``-g3``)
+   * Function mode does not require debug info but does not support stripped binaries
+
+Backends
+-----------------------------------
+
+There are two backends to choose from: ``perf`` and ``timer``. 
+They are used to record the samples required to calculate the virtual speedup. 
+Both backends interrupt each thread 1000 times per second (of CPU-time) to apply the virtual speed-ups.
+The difference between each backend is how the samples are recorded.
+There are three key differences between the two backends:
+
+* the ``perf`` backend requires Linux Perf and elevated security priviledges
+* the ``perf`` backend interrupts the application less frequently whereas the ``timer`` backend 
+  interrupts the application 1000 times per second of realtime
+* the ``timer`` backend has less accurate call stacks due to instruction pointer skid
+
+In general, the ``perf`` backend is preferred over the ``timer`` backend when sufficient 
+security priviledges permit its usage.
+If ``OMNITRACE_CAUSAL_BACKEND`` is set to ``auto``, Omnitrace falls back 
+to using the ``timer`` backend only if
+the ``perf`` backend fails. If ``OMNITRACE_CAUSAL_BACKEND`` is 
+set to ``perf`` and using this backend fails, Omnitrace aborts.
+
+Instruction pointer skid
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Instruction pointer (IP) skid measures how many instructions run after the event of interest
```

Comment:
@dgaliffiAMD This is the reworked definition. Do you think it's clear now, or should we add a diagram or more explanation.?

Reply:
I think your change works. Thank you.